Services/Data/Providers/DatabaseOrdersProvider.cs

```diff
@@ -26,13 +26,27 @@ public async Task<IEnumerable<Order>> GetAllOrders()
 
                 foreach (var orderFromDb in ordersDTO)
                 {
-                    var dishes = orderFromDb.OrderItems.Select(oi => new Dish(oi.DishId, oi.Dish.Name, oi.Dish.ImagePath, oi.Dish.Recipe, oi.Dish.Ingredients));
+                    var dishes = orderFromDb.OrderItems.Select(oi => new Dish(oi.DishId, oi.Dish.Name, oi.Dish.ImagePath, oi.Dish.Recipe, oi.IsCompleted, oi.Dish.Ingredients));
 
                     orders.Add(new Order(dishes, orderFromDb.CreatedOn, orderFromDb.Id));
                 }
 
                 return orders;
             }
         }
+
+        public async Task CompleteDish(int dishId, int orderNumber)
+        {
+            using (var dbContext = _dbContextFactory.CreateDbContext())
+            {
+                var dishesToChangeCompletedState = await dbContext.OrderItems.Where(oi => oi.DishId == dishId && oi.OrderId == orderNumber).ToListAsync();
+
+
+                dishesToChangeCompletedState.ForEach(oi => oi.IsCompleted = true);
```

Comment:
It's not clear from the task, but the dishes are marked ready one at a time.
I'll fix it in the task.